### PR TITLE
:letter: add some readme info

### PR DESCRIPTION
**Pull request by beseven**

### CODE REVIEW

**Comment by c2s-dev** [comment]
:rocket: deployment of gp-finder-pr-87 in nhsuk-dev succeeded (http://gp-finder-pr-87.dev.beta.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployment of gp-finder-pr-87 in nhsuk-dev succeeded (http://gp-finder-pr-87.dev.beta.nhschoices.net)

File: docker-compose-tests.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ services:
       - gp-finder-travis-tests-node_modules:/code/node_modules
     environment:
       COVERALLS_REPO_TOKEN: ${COVERALLS_REPO_TOKEN}
-      NODE_ENV: development
+      NODE_ENV: ${NODE_ENV-development}
```
Comment (st3v3nhunt) [inline docker-compose-tests.yml]: Does the variable want setting when the value is either unset or empty or just when it is unset?
See docs for more info: 
https://docs.docker.com/compose/compose-file/compose-file-v2/#variable-substitution

File: docker-compose.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ services:
     build:
       context: .
       args:
-        NODE_ENV: development
+        NODE_ENV: ${NODE_ENV-development}
```
Comment (st3v3nhunt) [inline docker-compose.yml]: Does the variable want setting when the value has is either unset or empty or just when it is unset?
See docs for more info: 
https://docs.docker.com/compose/compose-file/compose-file-v2/#variable-substitution

File: docker-compose-tests.yml [hunk 3]
```diff
@@ -10,7 +10,7 @@ services:
     volumes:
       - gp-finder-tests-node_modules:/code/node_modules
     environment:
-      NODE_ENV: test
+      NODE_ENV: ${NODE_ENV-test}
```
Comment (st3v3nhunt) [inline docker-compose-tests.yml]: Does the variable want setting when the value has is either unset or empty or just when it is unset?
See docs for more info: 
https://docs.docker.com/compose/compose-file/compose-file-v2/#variable-substitution

File: README.md [hunk 4]
```diff
@@ -24,12 +24,14 @@ the application is being run. This is best practice as described by
 
 Start by cloning the repo and all submodules i.e. `git clone https://github.com/nhsuk/gp-finder.git && cd gp-finder/ && git submodule update --init --recursive`
 
-Run the application with Docker via `docker-compose up --build --force-recreate`. This will build an image based on the code in the current working directory and start it running. It will most likely be available locally on http://localhost:3000 as long as the port exposed in `docker-compose.yml` is `3000`.
+You can run the application with Docker via `docker-compose up --build --force-recreate`. This will build an image based on the code in the current working directory and start it running. It will most likely be available locally on http://localhost:3000 as long as the port exposed in `docker-compose.yml` is `3000`.
```
Comment (st3v3nhunt) [inline README.md]: I'd suggest editing this entire section to cover the use of the scripts. The details of what is happening is contained in the scripts so it shouldn't (?) need to be duplicated within the readme. Possibly worth adding a comment or two into the scripts if that makes sense to do so.

File: docker-compose-tests.yml (hunk 3 above)
Comment (beseven) [inline docker-compose-tests.yml]: @st3v3nhunt I see your point about the `test` bit. Is there a reason we use `development` for Travis though? The thing I'm trying to fish at is - can we make the 2 test scripts as similar as possible?

Comment (st3v3nhunt) [inline docker-compose-tests.yml]: Additional logging?

File: docker-compose-tests.yml (hunk 1 above)
Comment (beseven) [inline docker-compose-tests.yml]: To keeo it compatible with this: https://docs.rancher.com/rancher/v1.5/en/cli/variable-interpolation/#variable-interpolation-formats and my understanding I would say stick to only unset. I think I need to understand how the envs are used more if we were to have a `start-prod` /cc @adamgraves-choices

File: README.md (hunk 4 above)
Comment (beseven) [inline README.md]: Would also be good to decide which scripts are we going with the ones here in `gp-finder` or the ones in `profiles` before we write down how they work/document them?

**Comment by c2s-dev** [comment]
:rocket: deployment of gp-finder-pr-87 in nhsuk-dev succeeded (http://gp-finder-pr-87.dev.beta.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployment of gp-finder-pr-87 in nhsuk-dev succeeded (http://gp-finder-pr-87.dev.beta.nhschoices.net)

File: scripts/test [hunk 5]
```diff
@@ -1,3 +1,4 @@
 #!/bin/bash
+# -p flag will create a different network for the test container so it won't interfere with the app container
```
Comment (st3v3nhunt) [inline scripts/test]: This comment doesn't match the description given in the docker compose documentation. See:

* https://docs.docker.com/compose/reference/overview/
* https://docs.docker.com/compose/reference/envvars/#compose-project-name

File: README.md [hunk 6]
```diff
@@ -22,14 +22,12 @@ the application is being run. This is best practice as described by
...
+Please run the `./scripts/bootstrap` script first and follow the instructions.
 
-When finished with the application `docker-compose down -v` should be run to shutdown all services including volumes. This is the correct way to close down resource used by the `up` command.
+For application specific development, start by cloning the repo and all submodules i.e. `git clone https://github.com/nhsuk/gp-finder.git && cd gp-finder/ && git submodule update --init --recursive`
 
-It is good practice to run the tests in a Docker container. To do this run `docker-compose -p gp-finder-test -f docker-compose-tests.yml up --build --force-recreate tests`. A new container will be started where the tests will run and rerun when changes are made to the source code.
-When finished with the tests run `docker-compose -p gp-finder-test -f docker-compose-tests.yml down -v` to tidy up.
+You can run the application using `./scripts/start`. The application will be available locally on `http://localhost:3000`.
```
Comment (st3v3nhunt) [inline README.md]: The script could be linked to e.g. [`./scripts/start`](scripts/start)

File: README.md [hunk 7]
```diff
@@ -22,14 +22,12 @@ the application is being run. This is best practice as described by
...
```
Comment (st3v3nhunt) [inline README.md]: Same as above comment about linking to the script, if that was wanted

File: scripts/README.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
+# Scripts (too rule them all)
+
+The scripts in these folder are meant to help the user/contributor to get up and running with the application fast.
+
+`bootstrap` - will point you towards the setup you will need before picking up this application
```
Comment (st3v3nhunt) [inline scripts/README.md]: As with the main `README`, this script, and the ones below could be linked to - if that was wanted?

File: scripts/test (hunk 5 above)
Comment (beseven) [inline scripts/test]: This is why I was talking about the network: https://docs.docker.com/compose/networking/ 

Note: Your app’s network is given a name based on the “project name”, which is based on the name of the directory it lives in. You can override the project name with either the --project-name flag or the COMPOSE_PROJECT_NAME environment variable.

**Comment by c2s-dev** [comment]
:rocket: deployment of gp-finder-pr-87 in nhsuk-dev succeeded (http://gp-finder-pr-87.dev.beta.nhschoices.net)

File: scripts/README.md [hunk 9]
```diff
@@ -2,10 +2,11 @@
 
 The scripts in these folder are meant to help the user/contributor to get up and running with the application fast.
 
-`bootstrap` - will point you towards the setup you will need before picking up this application
+[`bootstrap`](scripts/bootstrap)- will point you towards the setup you will need before picking up this application
```
Comment (st3v3nhunt) [inline scripts/README.md]: The path for `scripts` doesn't need to be here. Just [`bootstrap`](bootstrap) will suffice.
Same for the 3 links below

Comment (beseven) [inline scripts/README.md]: On it

**Pull request merged**